diag/helpers.go

```diff
@@ -6,10 +6,11 @@ import "fmt"
 // as the most common use case in Go will be handling a single error
 // returned from a function.
 //
-//   if err != nil {
-//     return diag.FromErr(err)
-//   }
+//   return diag.FromErr(err)
```

Comment:
```suggestion
//   if err != nil {
//     return diag.FromErr(err)
//   }
```

Reply:
Going to leave this as the common usage is to nil check errors first, but I think its a good fix.